src/main/java/net/rubygrapefruit/platform/internal/jni/AbstractFileEventFunctions.java

```diff
@@ -44,6 +45,84 @@ public AbstractWatcherBuilder(BlockingQueue<FileWatchEvent> eventQueue) {
          * @see FileWatcher#startWatching(Collection)
          */
         public abstract FileWatcher start() throws InterruptedException;
+
+        /**
+         * Starts the file watcher with a thread consuming the events and
+         * feeding them to the given {@link FileWatchEvent.Handler}.
+         *
+         * The started thread will terminate once the returned
+         * {@link FileWatcher} is {@link FileWatcher#close() closed}.
+         */
+        public FileWatcher startWithHandler(FileWatchEvent.Handler handler) throws InterruptedException {
+            return startWithHandler(new ThreadConfigurator() {
+                @Override
+                public void configure(Thread processorThread) {
+                    // Do nothing
+                }
+            }, handler);
+        }
+
+        /**
+         * Starts the file watcher with a thread consuming the events and
+         * feeding them to the given {@link FileWatchEvent.Handler}.
+         *
+         * The started thread will terminate once the returned
+         * {@link FileWatcher} is {@link FileWatcher#close() closed},
+         * or when it is {@link Thread#interrupt() interrupted}.
+         *
+         * Allows the configuration of the thread created via the given
+         * {@link ThreadConfigurator} before it is started.
+         */
+        public FileWatcher startWithHandler(ThreadConfigurator threadConfigurator, final FileWatchEvent.Handler handler) throws InterruptedException {
```

Comment:
I find it strange that the methods which start a thread still receive a blocking queue. It seems like you shouldn't care in that case.

Moreover, the method doesn't document what happens if the handler throws an exception. Currently, the thread would stop and all the remaining file events will overflow the queue. Shouldn't the watcher be stopped as well when there is an error?

All in all, it seems to me that the methods starting the thread shouldn't be in the builder, but a wrapper around the builder. Can we remove them from this PR and do those changes in a follow up if we want to support creating the consuming thread from native platform?

We should discuss how we stop the watcher and the consuming thread in an orderly fashion and how to handle the different kind of errors which can arise.

Reply:
I'm fine with moving the consuming thread to another PR, though I think you'd still want to be able to customize the queue instance used.